spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/MappedToGettableDataConverter.scala

```diff
@@ -0,0 +1,187 @@
+package com.datastax.spark.connector.writer
+
+import scala.language.existentials
+import scala.reflect.runtime.universe._
+import scala.util.control.NonFatal
+
+import com.datastax.spark.connector.util.{Symbols, ReflectionUtil}
+import com.datastax.spark.connector.{TupleValue, UDTValue, ColumnRef}
+import com.datastax.spark.connector.cql.StructDef
+import com.datastax.spark.connector.mapper.{DefaultColumnMapper, TupleColumnMapper, JavaBeanColumnMapper, ColumnMapper}
+import com.datastax.spark.connector.types.{TupleType, MapType, SetType, ListType, ColumnType, TypeConverter}
+
+private[connector] object MappedToGettableDataConverter {
+
+  def apply[T : TypeTag : ColumnMapper]
+      (struct: StructDef, columnSelection: IndexedSeq[ColumnRef]): TypeConverter[struct.ValueRepr] =
+
+    new TypeConverter[struct.ValueRepr] {
+
+      /** To determine which default column mapper to use for UDT fields.
+        * If this class is a Java bean, then the UDT fields will be mapped with
+        * the `JavaBeanColumnMapper` */
+      private val isJavaBean =
+        implicitly[ColumnMapper[T]].isInstanceOf[JavaBeanColumnMapper[_]]
+
+      private val columnMap =
+        implicitly[ColumnMapper[T]].columnMapForWriting(struct, columnSelection)
+
+      /** Returns the column mapper associated with the given type.
+        * Used to find column mappers for UDT columns.
+        * For tuples, returns [[com.datastax.spark.connector.mapper.TupleColumnMapper TupleColumnMapper]],
+        * for Java beans uses
+        * [[com.datastax.spark.connector.mapper.JavaBeanColumnMapper JavaBeanColumnMapper]],
+        * and for everything else uses
+        * [[com.datastax.spark.connector.mapper.DefaultColumnMapper DefaultColumnMapper]] */
+      private def columnMapper[U: TypeTag]: ColumnMapper[U] = {
+        val tpe = TypeTag.synchronized(typeTag[U].tpe)
+        if (ReflectionUtil.isScalaTuple(tpe))
+          new TupleColumnMapper[U]
+        else if (isJavaBean)
+          new JavaBeanColumnMapper[U]()(ReflectionUtil.classTag[U])
+        else
+          new DefaultColumnMapper[U]
+      }
+
+      /** For these types we don't want to use column mappers */
+      private val dynamicStructRepresentations = Set(
+        TupleValue.Symbol.fullName,
+        UDTValue.Symbol.fullName,
+        "com.datastax.spark.connector.japi.TupleValue",
```

Comment:
Could be brittle if changes occur in package structure for japi. Will compiler catch that if that happens?